cmd/swarmctl/service/common.go

```diff
@@ -13,28 +13,63 @@ func getService(ctx context.Context, c api.ControlClient, input string) (*api.Se
 	rg, err := c.GetService(ctx, &api.GetServiceRequest{ServiceID: input})
 	if err != nil {
 		// If any error (including NotFound), ListServices to match via ID prefix and name prefix.
-		rl, err := c.ListServices(ctx,
-			&api.ListServicesRequest{
-				Filters: &api.ListServicesRequest_Filters{
-					NamePrefixes: []string{input},
-				},
-			},
-		)
+		service, err := getServiceByName(ctx, c, input)
 		if err != nil {
-			return nil, err
+			service, err = getServiceByPrefixedID(ctx, c, input)
+			if err != nil {
+				return nil, err
+			}
 		}
+		return service, nil
+	}
 
-		if len(rl.Services) == 0 {
-			return nil, fmt.Errorf("service %s not found", input)
-		}
+	return rg.Service, nil
+}
 
-		if l := len(rl.Services); l > 1 {
-			return nil, fmt.Errorf("service %s is ambiguous (%d matches found)", input, l)
-		}
+func getServiceByPrefixedID(ctx context.Context, c api.ControlClient, input string) (*api.Service, error) {
+	rl, err := c.ListServices(ctx,
+		&api.ListServicesRequest{
+			Filters: &api.ListServicesRequest_Filters{
+				IDPrefixes: []string{input},
+			},
+		},
+	)
+	if err != nil {
+		return nil, err
+	}
 
-		return rl.Services[0], nil
+	if len(rl.Services) == 0 {
+		return nil, fmt.Errorf("service %s not found", input)
 	}
-	return rg.Service, nil
+
+	if l := len(rl.Services); l > 1 {
+		return nil, fmt.Errorf("service %s is ambiguous (%d matches found)", input, l)
+	}
+
+	return rl.Services[0], nil
+}
+
+func getServiceByName(ctx context.Context, c api.ControlClient, input string) (*api.Service, error) {
```

Comment:
`getServiceByName` and `getServiceByPrefixedID` are only different in the filters. Do you think it better to define `getServiceByFilters` and pass in filters as parameter?